server/channels/app/webhook.go

```diff
@@ -664,10 +664,10 @@ func (a *App) HandleIncomingWebhook(c request.CTX, hookID string, req *model.Inc
 		return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.disabled.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	hchan := make(chan store.StoreResult, 1)
+	hchan := make(chan store.GenericStoreResult[*model.IncomingWebhook], 1)
 	go func() {
 		webhook, err := a.Srv().Store().Webhook().GetIncoming(hookID, true)
-		hchan <- store.StoreResult{Data: webhook, NErr: err}
+		hchan <- store.GenericStoreResult[*model.IncomingWebhook]{Data: webhook, NErr: err}
 		close(hchan)
 	}()
```

Comment:
I wonder if we need the concurrency at all here, given that in L687 we wait for the result anyway.

Reply:
A nice question I'm not ready to answer yet =)

Reply:
Probably not a significant improvement, but we have many examples 🤷